File: ex4/mad-lib.rb
# exercise 4
# ruby
# function plus variable interpellation

def question(prompt)
    puts prompt
    output = gets.chomp
    return output
end
    
noun = question("What singular noun?")
adjective = question("What adjective?")
verb = question("What singular verb \(e.g. eats\) ?")
adverb = question("What adverb?")

madlib = "The #{adjective} #{noun} #{verb} #{adverb}."
puts madlib

File: ex4/mad-lib.py
# ISM 2
# Exercise 4 (Hogan/Mad Lib)
# Python
# modified: function + substitution

prompt = "What's a good "

def question( prompt, part ):
    part = raw_input(prompt + part + "? ")
    return part



noun = question( prompt, "noun")
verb = question( prompt, "verb")
adjective = question( prompt, "adjective" )
adverb = question( prompt, "adverb" )

madlib = "Does your %s %s %s %s?" % (adjective, noun, verb, adverb)

print madlib




